### PR TITLE
allow amp-ad to be blocked by amp-user-notification even when no amp-ad scope exists

### PR DESCRIPTION
**Pull request by erwinmombay**
Closes https://github.com/ampproject/amphtml/issues/2633

### CODE REVIEW

**Comment by erwinmombay** [comment]
@cramforce currently amp-ad's don't get blocked by amp-user-notification if the ad type is not in https://github.com/ampproject/amphtml/blob/master/ads/_config.js#L113 , since no scope will exists.
we're ok with doing this right? 

still need to write tests, just wanted to make sure this was valid to do

**Comment by cramforce** [comment]
Yeah, this is a good change.

**Comment by erwinmombay** [comment]
@dvoytenko PTAL


/cc @cramforce

File: builtins/amp-ad.js [hunk 1]
```diff
@@ -300,12 +302,13 @@ export function installAd(win) {
           return;
         }
         let consent = Promise.resolve();
-        const consentId = this.element.getAttribute(
-            'data-consent-notification-id');
         if (consentId) {
           consent = userNotificationManagerFor(this.getWin()).then(service => {
             return service.get(consentId);
           });
+          if (!scope) {
```
Comment (dvoytenko) [inline builtins/amp-ad.js]: Move this outside of `if` to make it more readable for what happens with and without `scope` var.

File: builtins/amp-ad.js [hunk 2]
```diff
@@ -292,7 +292,9 @@ export function installAd(win) {
      */
     getAdCid_() {
       const scope = clientIdScope[this.element.getAttribute('type')];
-      if (!scope) {
+      const consentId = this.element.getAttribute(
+          'data-consent-notification-id');
+      if (!(scope || consentId)) {
         return Promise.resolve();
       }
       return cidForOrNull(this.getWin()).then(cidService => {
```
Comment (dvoytenko) [inline builtins/amp-ad.js]: When `!scope`, why do we need to query `cidFor`? Does it add any real cost on the other hand?

Comment (erwinmombay) [inline builtins/amp-ad.js]: so `userNotificationMananger` is still going to need `cidFor` eventually which is why i thought it was ok to do keep on doing this here.

**Comment by dvoytenko** [comment]
LGTM with one request.

File: builtins/amp-ad.js (hunk 1 above)
Comment (erwinmombay) [inline builtins/amp-ad.js]: done. have to do:

```js
if (!scope && consentId) {
  return consent;
}
```

**Pull request merged**